pkg/datapath/loader/netlink.go

```diff
@@ -141,3 +142,15 @@ func (l *Loader) DeleteDatapath(ctx context.Context, ifName, direction string) e
 
 	return nil
 }
+
+// HasDatapath returns true if the interface has a tc filter attached on the given direction.
+func (l *Loader) HasDatapath(ctx context.Context, ifName, direction string) (bool, error) {
+	args := []string{"filter", "show", "dev", ifName, direction}
+	cmd := exec.CommandContext(ctx, "tc", args...).WithFilters(libbpfFixupMsg)
+	output, err := cmd.Output(log, true)
+	if err != nil {
+		return true, fmt.Errorf("failed to get tc filter: %w", err)
```

Comment:
Is `true` the proper thing to return here along with the error?
If there was an error, there will be other issues down the road so I guess it does not matter much

Reply:
Good question. So in that case, we don't know if the device has a program attached or not. So the rationale for returning true here is that I think we want to _attempt_ the removal of the BPF program. Having that program still attached to the device could lead to more errors than attempting to remove an absent program (removing an absent program will just generate the inconsequential error and warnings above).

Reply:
Makes sense. Thanks for the explanation.

Reply:
That's a good point, when a function returns an error all other values should be ignored. Since this is the case I would suggest to check for the error type in the caller of this function with the `errors.Is` function and perform some logic based on that information.